src/test/java/uk/gov/hmcts/ccd/domain/service/common/AccessControlServiceTest.java

```diff
@@ -2622,6 +2622,90 @@ void shouldRemoveFieldIfRelevantAclMissingWithMultipartyFixForAllEvents() {
             assertEquals(0, eventTrigger.getCaseFields().size());
         }
 
+        @Test
+        @DisplayName("Should remove field if relevant acl missing with multiparty fix for all case types and events")
+        void shouldRemoveFieldIfRelevantAclMissingWithMultipartyFixForAllCaseTypesAndEvents() {
+            Mockito.when(applicationParams.isMultipartyFixEnabled()).thenReturn(true);
+            Mockito.when(applicationParams.getMultipartyCaseTypes()).thenReturn(singletonList("*"));
+            Mockito.when(applicationParams.getMultipartyEvents()).thenReturn(singletonList("*"));
+
+            final CaseTypeDefinition caseType = newCaseType()
+                .withField(newCaseField()
+                    .withFieldType(aFieldType().withType("Text").build())
+                    .withId("Addresses")
+                    .withAcl(anAcl()
+                        .withRole(ROLE_NOT_IN_USER_ROLES)
+                        .withCreate(true)
+                        .withUpdate(true)
+                        .withRead(true)
+                        .build())
+                    .build())
+                .build();
+
+            CaseUpdateViewEvent caseEventTrigger = newCaseUpdateViewEvent()
+                .withField(
+                    aViewField()
+                        .withFieldType(aFieldType().withType("Text").build())
+                        .withId("Addresses")
+                        .build())
+                .build();
+
+            assertEquals(1, caseEventTrigger.getCaseFields().size());
+
+            CaseUpdateViewEvent eventTrigger = accessControlService.setReadOnlyOnCaseViewFieldsIfNoAccess(
+                CASE_TYPE_ID,
+                CASE_REFERENCE,
+                EVENT_ID,
+                caseEventTrigger,
+                caseType.getCaseFieldDefinitions(),
+                ACCESS_PROFILES,
```

Comment:
could we not re-use the test logic in a common method to avoid code duplications?

Reply:
Removed redundant code changes